### PR TITLE
init contract - reg/invoke

### PR DESCRIPTION
**Pull request by boqiu**
Smart contract component startup. Implement reg/invoke with mock service.

### CODE REVIEW

File: contract/external.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+* @copyright defined in go-seele/LICENSE
+ */
+package contract
+
+import (
+	"github.com/seeleteam/go-seele/crypto"
+)
+
+const (
+	txTypeContractReg int = iota
```
Comment (WTer) [inline contract/external.go]: what's iota?

Comment (boqiu) [reply contract/external.go]: auto increase value start with 0, this is golang syntax for enum definition.

File: contract/external.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	txTypeContractCall
+	txTypeContractDel
+)
+
+// Account wraps the information of smart contract account.
+type Account struct {
+	AccountAddress string
+	CodeAddress    string
+	State          []byte
```
Comment (WTer) [inline contract/external.go]: add some comments for these members.

Comment (WTer) [reply contract/external.go]: I suppose State's type is something like enum

File: contract/external.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// Transaction wraps the smart contract transaction.
+type Transaction struct {
+	from    string
+	to      string
+	txType  int
+	payload []byte
+	sig     *crypto.Signature
```
Comment (WTer) [inline contract/external.go]: seems rename sig as sign should be better and common

File: contract/external.go (hunk 2 above)
Comment (boqiu) [reply contract/external.go]: All the code in external.go are temp, so please ignore the comment.

File: contract/external.go (hunk 3 above)
Comment (boqiu) [reply contract/external.go]: All the code in external.go are temp, so please ignore the comment.

File: contract/handler.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		log.Error("Transaction signature is invalid, and will not be archived into blockchain.")
+		return
+	}
+
+	switch tx.txType {
+	case txTypeContractReg:
+		onRegisterContract(tx)
+	case txTypeContractCall:
+		onInvokeContract(tx)
+	case txTypeContractDel:
```
Comment (WTer) [inline contract/handler.go]: add default case

File: contract/external.go (hunk 2 above)
Comment (boqiu) [reply contract/external.go]: The state should be of type object, but use []byte for serialization.

File: contract/handler.go (hunk 4 above)
Comment (boqiu) [reply contract/handler.go]: unnecessary, do not handle not supported tx type.

**Pull request merged**